packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -235,13 +275,18 @@ class _HourControl extends StatelessWidget {
     final TextStyle hourStyle = fragmentContext.mode == _TimePickerMode.hour
         ? fragmentContext.activeStyle
         : fragmentContext.inactiveStyle;
+    final String formattedHour = localizations.formatHour(
+      fragmentContext.selectedTime,
+      alwaysUse24HourFormat: MediaQuery.of(context).alwaysUse24HourFormat,
+    );
 
     return new GestureDetector(
       onTap: Feedback.wrapForTap(() => fragmentContext.onModeChange(_TimePickerMode.hour), context),
-      child: new Text(localizations.formatHour(
-        fragmentContext.selectedTime,
-        alwaysUse24HourFormat: MediaQuery.of(context).alwaysUse24HourFormat,
-      ), style: hourStyle),
+      child: new Semantics(
+        selected: fragmentContext.mode == _TimePickerMode.hour,
+        hint: localizations.timePickerHourModeAnnouncement,
```

Comment:
The hint usually tells users what interactions have what effect. So this one should probably say "double tap to select hours" instead of just "select hours".

Reply:
If it's not reflected in the docs for hint, we should add it there.

Reply:
Hmm, this only announces that the time picker changed mode to either hour or minute. There is no one action that will allow you to select hours, as there are many controls on the screen. This is consistent with what Android does. It just says "select hours"/"select minutes". Although perhaps in a future iteration of the time picker we could improve the situation.